### ISSUE TITLE
OpenCLContext::Create(method:4):CL_INVALID_PLATFORM when run OpenCLResampler

### ISSUE THREAD

**Issue by pypi20200320** [SuperElastix/elastix#1084]
Hi,
I am trying to compile elastix with ELASTIX_USE_OPENCL: ON to make use of the GPU. I followed the steps at [https://simpleelastix.readthedocs.io/GettingStarted.html#manually-building-on-linux] to build ITK with ITK_USE_GPU: OFF (like it was suggested here: https://github.com/SuperElastix/elastix/issues/15).
To build elastix I used the following variables:

ELASTIX_INCLUDE_DIR: include
ELASTIX_USE_OPENCL: ON
set(ITK_DIR "/data/result/XXX/ITK/itk_build")
set(OPENCL_INCLUDE_DIRS "/usr/local/cuda-11.0/include")
set(OPENCL_LIBRARIES "/usr/lib/x86_64-linux-gnu/libOpenCL.so")

To configure the above variables followed by the "make -j4" command inside the /elastix/build directory. I used the ITK branch "release" and elastix branch "main".

And Compiling finished whit nothing error. But when i run elastix with OpenCLResampler,  it runed completely the same as defaultResampler, and just said "Details: OpenCLContext::Create(method:4):CL_INVALID_PLATFORM

  OpenCL processing in elastix is disabled." And the whole log is attached below. 

[the info.txt](https://github.com/SuperElastix/elastix/files/14743030/the.info.txt)

Has anyone encountered the same issue?

**Comment by N-Dekker** [comment]
Thanks for reporting this issue @pypi20200320 I guess the error occurs in `OpenCLContext::CreateContext`, when calling `clCreateContext`: 

https://github.com/SuperElastix/elastix/blob/23aad88d61d18a93ea8616152f161f91f04125a5/Common/OpenCL/ITKimprovements/itkOpenCLContext.cxx#L532

But honestly I don't really know  🤷

**Comment by pypi20200320** [comment]
@N-Dekker Thank you very much for your quick response. I suspect that the issue lies with my own installation of OpenCL. I am currently working on resolving it, and I will provide an update if there is any new information.

**Comment by pypi20200320** [comment]
@N-Dekker I apologize for the delayed update. I have resolved the issue, and it turns out that the error was caused by a faulty installation of OpenCL. 
In the end, I installed OpenCL on Ubuntu 20.04 and followed the steps outlined in the documentation at [https://simpleelastix.readthedocs.io/GettingStarted.html#manually-building-on-linux]. I successfully compiled the code and tested the OpenCLResampler, which indeed demonstrated significant acceleration. Thanks for your assistance.
